### ISSUE TITLE
floats spawn "out of bounds"

### ISSUE THREAD

**Issue by Gusser93** [hyprwm/Hyprland#669]
## Steps to reproduce

1. Have a similar monitor setup (I guess?)
2. Open something like a subwindow, for example the settings of NewsFlash, which opens as a floating window

## Expected outcome

- Subwindow is visible, ether in the middle of the spawning window or of the workspace

## Noted outcome

- The Subwindow spawns outside of the monitor of the workspace.

# Suggestion

Allow to ether spawn such modals/settings/subwindows in the center of the spawning window or at the center of the workspace of the spawning window.

# Monitor config
```
Monitor eDP-1 (ID 0):
	1920x1080@60.020000 at 2300x1457
	active workspace: 2 (2)
	reserved: 0 31 0 0
	scale: 1.60
	transform: 0
	focused: yes

Monitor HDMI-A-1 (ID 1):
	1920x1080@60.000000 at 380x960
	active workspace: 8 (8)
	reserved: 0 31 0 0
	scale: 1.00
	transform: 0
	focused: no

Monitor DP-2 (ID 2):
	1920x1080@60.000000 at 3500x120
	active workspace: 7 (7)
	reserved: 0 31 0 0
	scale: 1.00
	transform: 1
	focused: no
```
## Valid coordinates

| Monitor | upper left | bottom right | spawn coordinates | 
| -- | -- | -- | -- |
| left (ID 1) | 380, 960 | 2300, 2040 | 56, 50 |
| middle (ID 0) | 2300, 1457 | 3500, 2132 | 1976, 547 |
| right (ID 2) | 3500, 120 | 4580, 2040 | 3176, -790

## relevant config

```
monitor=eDP-1,preferred,2300x1457,1.6
monitor=HDMI-A-1,preferred,380x960,1
monitor=DP-2,preferred,3500x120,1
monitor=,preferred,auto,1
monitor=DP-2,transform,1

windowrule=float,^(org.gnome.Calendar)$ # only windowrule
```

# Images
## Left monitor

Einstellungen (german for settings) spawns at 56,50
![2022-09-15_15-54_screenshot_rect](https://user-images.githubusercontent.com/12176313/190431435-3d505fc9-8c34-46dd-a983-7dedbad01c94.png)

## Middle monitor

Einstellungen spawns at 1976,547
![2022-09-15_15-53_screenshot_rect](https://user-images.githubusercontent.com/12176313/190431499-615db0c2-f782-4005-b39d-c9dad1dc4fca.png)

## Right monitor

Einstellungen spawns at 3176, -790
![2022-09-15_15-52_screenshot](https://user-images.githubusercontent.com/12176313/190423324-ed40f6c8-def7-4b33-9b1e-c891d103dd25.png)

**Comment by Gusser93** [comment]
```
Hyprland, built from branch main at commit 928158bbfb906bd871738d60610d85634d1f8c7edirty (prevent missed surface size updates on fullscreen window close).
flags: (if any)
```
OS: Arch x86_64
Kernel: 5.15.67-1-lts
Package: [hyprland-bin](https://aur.archlinux.org/packages/hyprland-bin)

I can test the -git version Friday or Monday.

**Comment by vaxerski** [comment]
ye thats really old. test git

**Comment by Gusser93** [comment]
That did the trick, probably fixed by 7d4f0a3.
